clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -1278,5 +1275,12 @@ static class MemberInfo {
             this.memberId = Optional.empty();
             this.memberEpoch = Optional.empty();
         }
+
+        @Override
+        public String toString() {
+            return "MemberInfo{" + "memberId=" + memberId.orElse("undefined") +
+                    ", memberEpoch=" + (memberEpoch.isPresent() ? memberEpoch : "undefined") + "}";
```

Comment:
`memberEpoch.get`?

Reply:
I would say we need to check if present. There could be no epoch/id when committing/fetching (if we haven't joined a group or left it. In those cases we would have empty here, so that no epoch/id is included in the OffsetFetch/OffsetCommit request).

Reply:
Agreed! And my point was "we should call get if it is present". Otherwise, the toString will include "Optional"

Reply:
I implemented this suggestion, I will note that when testing the method, there is no difference in how the string is printed whether using ```memberEpoch.get()``` or not

Reply:
> Agreed! And my point was "we should call get if it is present". Otherwise, the toString will include "Optional"

Oh I misread your point he he. But both on the same page in the end, great.